go/vt/vtgate/planbuilder/operators/horizon.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/plancontext"
+)
+
+// Horizon is an operator we use until we decide how to handle the source to the horizon.
```

Comment:
I am not sure I fully understand this. We add the `Horizon` operator into the tree until we have decided how the rest of the tree will do horizon planning?

Reply:
Looking at the operator test data, it feels like it encapsulates each node in the tree.

Reply:
Yes. We need to do the physical route planning before we do the horizon planning